app/notification_push.go

```diff
@@ -294,11 +359,40 @@ func (a *App) sendToPushProxy(msg model.PushNotification, session *model.Session
 		mlog.Info(fmt.Sprintf("Device was reported as removed for UserId=%v SessionId=%v removing push for this session", session.UserId, session.Id), mlog.String("user_id", session.UserId))
 		a.AttachDeviceId(session.Id, "", session.ExpiresAt)
 		a.ClearSessionCacheForUser(session.UserId)
+		err := errors.New("Device was reported as removed")
```

Comment:
nit: maybe a little in-line function for the 3 similar flows here?

Reply:
Yes, good point, I going to rewrite it

Reply:
Done